infra/main.tf

```diff
@@ -19,6 +19,19 @@ variable "batch_logs_bucket_storage_class" {}
 variable "gcp_region" {}
 variable "gcp_zone" {}
 variable "domain" {}
+variable "use_artifact_registry" { 
+  type = bool
+  description = "pull the ubuntu image from Artifact Registry. Otherwise, GCR"
```

Comment:
I feel like, rather than have this additional configuration, we should also switch to the artifact registry.  I'm happy for this to go in as is, but we should consider that @daniel-goldstein.